codegenerator/cli/npm/envio/src/ReorgDetection.res

```diff
@@ -402,31 +407,34 @@ module LastBlockScannedHashes: {
   /**
   Checks whether reorg has occured by comparing the parent hash with the last saved block hash.
   */
-  let rec hasReorgOccurredInternal = (
-    lastBlockScannedDataList,
-    ~firstBlockParentNumberAndHash: option<blockNumberAndHash>,
-  ) => {
-    switch (firstBlockParentNumberAndHash, lastBlockScannedDataList) {
-    | (Some({blockHash: parentHash, blockNumber: parentBlockNumber}), list{head, ...tail}) =>
-      if parentBlockNumber == head.blockNumber {
-        parentHash != head.blockHash
-      } else {
-        //if block numbers do not match, this is a dynamic contract case and should recurse
-        //through the list to look for a matching block or nothing to validate
-        tail->hasReorgOccurredInternal(~firstBlockParentNumberAndHash)
-      }
-    | _ => //If parentHash is None, either it's the genesis block (no reorg)
+  let rec hasReorgOccurredInternal = (lastBlockScannedDataList, ~reorgGuard: reorgGuard) => {
+    switch lastBlockScannedDataList {
+    | list{head, ...tail} =>
+      switch reorgGuard {
+      | {lastBlockScannedData} if lastBlockScannedData.blockNumber == head.blockNumber =>
+        lastBlockScannedData.blockHash != head.blockHash
```

Comment:
Check for `lastBlockScannedData` as well, so dynamic contract catch-up also detect reorgs